zkvm/src/vm.rs

```diff
@@ -644,8 +677,7 @@ where
         spacesuit::range_proof(
             self.delegate.cs(),
             r1cs::LinearCombination::from_iter(expr.terms),
-            // TBD: maintain the assignment for the expression and provide it here
-            None,
+            expr.assignment,
```

Comment:
If assignment is `Option<ScalarKind>` (aka `Number` aka `ScalarWitness`), then this can be changed using `Option.map`:

```suggestion
            match expr.assignment {
                None => None,
                Some(ScalarKind::Integer(i)) => Some(i),
                Some(ScalarKind::Scalar(_)) => return Err(VMError::InconsistentAssignment),
            }
```